contracts/LimitOrderSwap.sol

```diff
@@ -82,7 +82,7 @@ contract LimitOrderSwap is ILimitOrderSwap, Ownable, TokenCollector, EIP712, Ree
         uint256[] memory takerTokenAmounts = new uint256[](orders.length);
         uint256 wethToPay;
         address payable _feeCollector = feeCollector;
-        for (uint256 i = 0; i < orders.length; ++i) {
+        for (uint256 i = 0; i < orders.length; ) {
```

Comment:
nit: Shall we cache the `orders.length` and also other loops in this PR?

Reply:
Will do in another PR.